app/presenters/account_show_presenter.rb

```diff
@@ -34,10 +36,44 @@ def show_gpo_partial?
     user.gpo_verification_pending_profile?
   end
 
+  def show_ipp_partial?
+    user.pending_in_person_enrollment.present?
+  end
+
   def showing_any_partials?
+    show_password_reset_partial? ||
+      show_gpo_partial? || show_ipp_partial? || show_pii_partial?
+  end
+
+  def showing_alerts?
     show_service_provider_continue_partial? ||
-      show_password_reset_partial? ||
-      show_gpo_partial?
+      show_password_reset_partial?
+  end
+
+  def show_unverified?
+    show_password_reset_partial? &&
+      (!show_ipp_partial? || !show_gpo_partial?)
+  end
+
+  def find_sp_issuer
+    sp_name = 'APP_NAME'
+    user.profiles.each do |profile|
+      if profile.initiating_service_provider_issuer
+        sp_name = ServiceProvider.find_by(issuer: profile.initiating_service_provider_issuer).
+          friendly_name
+        break
+      end
+    end
+    return sp_name
+  end
+
+  def service_provider_or_app_name
```

Comment:
Because of the presenter name `AccountShowPresenter`, it's not obvious that service provider here refers specifically to the service provider associated with a pending profile. I think it might also simplify some of the complications of tracking down which profile we want to use by creating separate methods for each of the use-cases. Especially if we can guarantee that the methods would only be called when those conditions are met.

```rb
def pending_ipp_profile_service_provider
  user.pending_in_person_enrollment.initiating_service_provider.friendly_name
end

def pending_gpo_profile_service_provider
  user.gpo_verification_pending_profile.initiating_service_provider.friendly_name
end
```

Reply:
This was updated in ac9737e087.